integration/subxt-tests/src/cases/events.rs

```diff
@@ -0,0 +1,71 @@
+use contract_transcode::ContractMessageTranscoder;
+use parity_scale_codec::{Compact, Decode, Input};
+use sp_core::{crypto::AccountId32, hexdisplay::AsBytesRef};
+
+use crate::{Contract, DeployContract, Execution, WriteContract, API};
+use hex::FromHex;
+
+#[tokio::test]
+async fn case() -> anyhow::Result<()> {
+    let api = API::new().await?;
+
+    let mut c = Contract::new("./contracts/Events.contract")?;
+
+    c.deploy(&api, sp_keyring::AccountKeyring::Alice, 0, |t| {
+        t.encode::<_, &'static str>("new", []).unwrap_or_default()
+    })
+    .await?;
+
+    let rs = c
+        .call(&api, sp_keyring::AccountKeyring::Alice, 0, |t| {
+            t.encode::<_, &'static str>("emit_event", [])
+                .unwrap_or_default()
+        })
+        .await?;
+
+    assert_eq!(rs.len(), 4);
+
+    // TODO: currently event decoding is different than ink, as we can see in contract-transcode.
```

Comment:
Is this still the case? I can't remember having this as an issue with polkadot-js, but I might be wrong. What exactly is the difference you encountered? I'm wondering if we could fix it in solang.

Reply:
I have to do some testing. I remember it being an extra byte off for event_id or some sort.

Reply:
Found it.
https://github.com/paritytech/cargo-contract/blob/ce9afac78eb76aa1c78998e08861764619f5e8ff/crates/transcode/src/lib.rs#LL293C9-L293C50

This file strips one `Compact<u32>` where solang generated event doesn't have. If we manually prepend one into the Event.data field, the decoding will success.

Reply:
Right, thanks for digging this out.